### PR TITLE
[WFLY-8522] Slave host connecting to master

### PR DESCRIPTION
**Pull request by rjanik114**
Jira: https://issues.jboss.org/browse/WFLY-8522
Planning jira: https://issues.jboss.org/browse/EAP7-632

Adds a test with slave connecting to master, referencing AuthenticationContext, using DIGEST-MD5 and PLAIN mechanisms

I'm asking for review since I don't usually work with security or managed domains.

### CODE REVIEW

**Review by kabir — APPROVED** [review]
Approving that the paperwork is in order

**Review by darranl — CHANGES_REQUESTED** [review]
As this is going into a community repo we will need a WFLY issue creating and the commit message updated to reference this instead of an internal planning issue.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/SlaveHostControllerAuthenticationTestCase.java [hunk 1]
```diff
@@ -58,21 +31,29 @@
 import org.jboss.as.test.integration.security.common.VaultHandler;
 import org.jboss.dmr.ModelNode;
 import org.junit.AfterClass;
-import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
+import javax.security.auth.callback.CallbackHandler;
```
Comment (darranl) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/SlaveHostControllerAuthenticationTestCase.java]: Please don't re-order the imports - the static imports belong at the top.

File: testsuite/domain/src/test/resources/domain-configs/domain-elytron.xml [hunk 2]
```diff
@@ -0,0 +1,514 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (darranl) [inline testsuite/domain/src/test/resources/domain-configs/domain-elytron.xml]: If this test case is testing host controller authentication is a complete domain xml really required?

**Comment by darranl** [comment]
Also generally a minor issue is a couple of the classes are new but have copyright headers for 2013, if they are genuinely new these should be 2017.

**Comment by darranl** [comment]
@bstansberry are you able to give this a quick review please as adding a new domain mode test?

**Comment by rjanik114** [comment]
@darranl Thank you for the review. I fixed the issues you pointed out and pushed the changes.

File: testsuite/domain/src/test/resources/domain-configs/domain-elytron.xml (hunk 2 above)
Comment (rjanik114) [reply testsuite/domain/src/test/resources/domain-configs/domain-elytron.xml]: No it wasn't, thanks. I used the minimal domain config and domain-elytron.xml is now gone.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/SlaveHostControllerAuthenticationTestCase.java (hunk 1 above)
Comment (rjanik114) [reply testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/SlaveHostControllerAuthenticationTestCase.java]: Sorry, fixed my IDE setup. The ordering should be as in the rest of the project now.

**Pull request merged**